cirq/ops/pauli_sum_exponential.py

```diff
@@ -0,0 +1,131 @@
+        self._pauli_sum = pauli_sum
+
+    @property
+    def qubits(self) -> Tuple['cirq.Qid', ...]:
+        return self._pauli_sum.qubits
+
+    def _value_equality_values_(self) -> Any:
+        return (self._pauli_sum, self._exponent)
+
+    def with_qubits(self, *new_qubits: 'cirq.Qid') -> 'PauliSumExponential':
+        return PauliSumExponential(self._pauli_sum.with_qubits(*new_qubits),
+                                   self._exponent,
+                                   is_anti_hermitian=self._is_anti_hermitian)
+
+    def _resolve_parameters_(self,
+                             param_resolver: 'cirq.ParamResolverOrSimilarType'
+                            ) -> 'PauliSumExponential':
+        return PauliSumExponential(self._pauli_sum,
+                                   exponent=protocols.resolve_parameters(
+                                       self._exponent, param_resolver),
+                                   is_anti_hermitian=self._is_anti_hermitian)
+
+    @staticmethod
+    def _all_pauli_strings_commute_(pauli_sum: 'cirq.PauliSum') -> bool:
+        for x in pauli_sum:
+            for y in pauli_sum:
+                if not protocols.commutes(x, y):
+                    return False
+        return True
+
+    def __iter__(self) -> Iterator['cirq.PauliStringPhasor']:
+        for pauli_string in self._pauli_sum:
+            theta = pauli_string.coefficient * self._multiplier
+            theta *= self._exponent / np.pi
+            if isinstance(theta, complex):
+                theta = theta.real
+            yield pauli_string_phasor.PauliStringPhasor(
+                pauli_string.with_coefficient(1.0),
+                exponent_neg=-theta,
+                exponent_pos=theta)
+
+    def matrix(self) -> np.ndarray:
+        """Reconstructs matrix of self from underlying Pauli sum exponentials.
+
+        Raises:
+            ValueError: if exponent is parameterized.
+        """
+        if protocols.is_parameterized(self._exponent):
+            raise ValueError("Exponent should not parameterized.")
+        ret = 1
+        for pauli_string_exp in self:
+            ret = np.kron(ret, protocols.unitary(pauli_string_exp))
+        return ret
+
+    def _has_unitary_(self) -> bool:
+        return linalg.is_unitary(self.matrix())
+
+    def _unitary_(self) -> np.ndarray:
+        return self.matrix()
+
+    def __pow__(self, exponent: int) -> 'PauliSumExponential':
+        return PauliSumExponential(self._pauli_sum,
+                                   self._exponent * exponent,
+                                   is_anti_hermitian=self._is_anti_hermitian)
+
+    def __repr__(self) -> str:
+        class_name = self.__class__.__name__
+        return f'cirq.{class_name}({self._pauli_sum!r}, {self._exponent!r}, '\
+               f'is_anti_hermitian={self._is_anti_hermitian!r})'
+
+    def __str__(self) -> str:
+        coeff = '1.0' if self._multiplier != 1 else 'j'
```

Comment:
We could even remove 1.0?

```suggestion
        coeff = '' if self._multiplier != 1 else 'j'
```

Reply:
done